cobbler/settings/migrations/V3_3_0.py

```diff
@@ -369,4 +383,114 @@ def migrate(settings: dict) -> dict:
     # delete removed keys
     helper.key_delete("cache_enabled", settings)
 
+    # migrate stored cobbler collections
+    migrate_cobbler_collections("/var/lib/cobbler/collections/")
+
     return normalize(settings)
+
+
+def backup_dir(dir_path: str):
+    """
+    Copies the directory tree and adds a suffix ".backup.XXXXXXXXX" to it.
+
+    :param dir_path: The full path to the directory which should be backed up.
+    :raises FileNotFoundError: In case the path specified was not existing.
+    """
+    src = os.path.normpath(dir_path)
+    now_iso = datetime.datetime.now().isoformat()
+    copytree(dir_path, f"{src}.backup.{now_iso}")
+
+
+def migrate_cobbler_collections(collections_dir: str):
+    """
+    Manipulate the main Cobbler stored collections and migrate deprecated settings
+    to work with newer Cobbler versions.
+
+    :param collections_dir: The directory of Cobbler where the collections files are.
+    """
+    backup_dir(collections_dir)
+    for collection_file in glob.glob(
+        os.path.join(collections_dir, "**/*.json"), recursive=True
+    ):
+        data = None
+        with open(collection_file, encoding="utf-8") as _f:
+            data = json.loads(_f.read())
+
+        # null values to empty strings
+        for key in data:
+            if data[key] is None:
+                data[key] = ""
+
+        # boot_loader -> boot_loaders
+        if "boot_loader" in data:
+            data["boot_loaders"] = data.pop("boot_loader")
+
+        # next_server -> next_server_v4, next_server_v6
+        if "next_server" in data:
+            addr = data["next_server"]
+            if addr == "<<inherit>>":
+                data["next_server_v4"] = addr
+                data["next_server_v6"] = addr
+                data.pop("next_server")
```

Comment:
This was good before. A single address can never be an IPv4 AND an IPv6. You can only set one and let the other be on the default.

Reply:
I've not change this. This is the case when `addr` is `<<inherit>>`.

Reply:
Right. Sorry that was my bad.